marklogic-data-hub-central/ui/src/components/entities/matching/threshold-modal/threshold-modal.tsx

```diff
@@ -0,0 +1,412 @@
+      let editThreshold = props.editThreshold;
+      setNameValue(editThreshold['thresholdName']);
+      setActionType(editThreshold['action']);
+      if (editThreshold['action'] === 'custom') {
+        setUriValue(editThreshold['actionModulePath'])
+        if (editThreshold.hasOwnProperty('actionModuleNamespace')) {
+          setNamespaceValue(editThreshold['actionModuleNamespace'])
+        }
+        if (editThreshold.hasOwnProperty('actionModuleFunction')) {
+          setFunctionValue(editThreshold['actionModuleFunction'])
+        }
+      }
+    }
+  }, [JSON.stringify(props.editThreshold)]);
+
+  const handleInputChange = (event) => {
+    switch(event.target.id) {
+      case 'name-input':
+        if (event.target.value === '') {
+          setNameErrorMessage('A threshold name is required');
+        } else {
+          setNameErrorMessage('');
+        }
+        setNameValue(event.target.value);
+        break;
+
+      case 'uri-input':
+        if (event.target.value === '') {
+          setUriErrorMessage('A URI is required');
+        } else {
+          setUriErrorMessage('');
+        }
+        setUriValue(event.target.value);
+        break;
+
+      case 'function-input':
+        if (event.target.value === '') {
+          setFunctionErrorMessage('A function is required');
+        } else {
+          setFunctionErrorMessage('');
+        }
+        setFunctionValue(event.target.value);
+        break;
+
+      case 'namespace-input':
+        setNamespaceValue(event.target.value);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const closeModal = () => {
+    resetModal();
+    props.toggleModal(false);
+  };
+
+  const resetModal = () => {
+    setNameValue('');
+    setNameErrorMessage('');
+    setActionType(undefined);
+    setActionTypeErrorMessage('');
+    setUriValue('');
+    setUriErrorMessage('');
+    setFunctionValue('');
+    setFunctionErrorMessage('');
+    setNamespaceValue('');
+  };
+
+  const onSubmit = async (event) => {
+    event.preventDefault();
+    let nameErrorMessage = '';
+    let actionErrorMessage = '';
+    let thresholdName = nameValue || '';
+    if (thresholdName === '') {
+      nameErrorMessage = 'A threshold name is required';
+    }
+
+    if (actionType === '' || actionType === undefined) {
+      actionErrorMessage = 'An action is required';
+    }
+    switch(actionType) {
+      case 'merge':
+      case 'notify':
+        {
+
+          if (actionErrorMessage === '' && nameErrorMessage === '' && Object.keys(props.editThreshold).length === 0) {
+            let newThreshold: Threshold = {
+              thresholdName,
+              action: actionType,
+              score: 0
+            };
+
+            let newStepArtifact: MatchingStep = curationOptions.activeStep.stepArtifact;
+            let duplicateNames = newStepArtifact.thresholds.filter( threshold => threshold.thresholdName === thresholdName);
```

Comment:
We need to check for duplicate names when the action is custom as well.

Reply:
Fixed this and made UX change to delete threshold confirmation